### ISSUE TITLE
An option to FinalResult which would skip all the deallocations

### ISSUE THREAD

**Issue by FredSRichardson** [alphacep/vosk-api#1037]
This is really just a feature suggestion to add batch mode segment based decoding to the Recognizer.

Implementing this myself, I ran into the following problems:
* When you call FinalResults() to get the final results the call deallocates a bunch of things including the decoder which has to be reinitialized now between each segment adding unnecessary overhead.
* There is no way to call InitState() which means word offset times are adjust by however much data you have decoded up to that point.

I would suggest adding an option to FinalResult() which would skip all the deallocations and an option to Result() and FinalResult() which does not adjust start/end times to support batch mode processing.  It would then be up to the user to periodically reallocate the decoder (for example to avoid memory leaks etc.).

**Comment by nshmyrev** [comment]
If you want to reallocate the decoder you simply reallocate the decoder. I don't see any issue. There is no overhead, decoder object is lightweight.

**Comment by FredSRichardson** [comment]
Yeah, that's not what I saw.  In Python, I added a step deallocating the decoder and reallocating between each segment and this ran significantly slower.

**Comment by nshmyrev** [comment]
And if you just feed data after FinalResult? It should also work.

**Comment by FredSRichardson** [comment]
Yes, this works, bit you are taking an unecessary hit deallocating and
reallocating the decoder.  Also you have a time offset added to each word
start time.  It's just not ideal for segment based processing though it is
great for streaming processing.
